optional-clustering-analysis/clustering-calculations.R

```diff
@@ -169,47 +176,128 @@ nn_range <- define_nn_range(opt$nearest_neighbors_min,
+  # Purpose: Calculate the cluster validity stats for each provided `cluster_type`
+  # across n range of parameters
+  #
+  # Args:
+  #   sce: clustered SingleCellExperiment object
+  #   nn_range: the range of numeric parameters used to test for clustering
+  #   cluster_type: method used for clustering
+  #   second_cluster_type: additional method used for clustering
+  
+  # Check the cluster validity stats for each of the clusters in the SCE object
+  # and return stats in a data frame
+  validity_stats_df <- create_metadata_stats_df(sce,
+                                                nn_range,
+                                                cluster_type)
+  
+  # If a second cluster type is provided, check the cluster validity stats for
+  # each of the clusters in the SCE object associated with the second cluster type
+  if(!is.null(second_cluster_type)){
+    second_validity_stats_df <- create_metadata_stats_df(sce,
+                                                         nn_range,
+                                                         second_cluster_type)
+    validity_stats_df <- second_validity_stats_df %>%
+      dplyr::bind_rows(validity_stats_df)
+  }
+}
 
-# Check the cluster validity stats for each of the clusters in the SCE object
-# and return stats in a data frame
-validity_stats_df <- create_metadata_stats_df(sce, 
-                                              nn_range, 
-                                              opt$cluster_type)
+cluster_stability_stats_wrapper_function <- function(sce, nn_range, cluster_type, second_cluster_type = NULL, ...) {
+  # Purpose: Calculate the cluster stability stats for each provided `cluster_type`
+  # across n range of parameters
+  #
+  # Args:
+  #   sce: clustered SingleCellExperiment object
+  #   nn_range: the range of numeric parameters used to test for clustering
+  #   cluster_type: method used for clustering
+  #   second_cluster_type: additional method used for clustering
+
+  # Check the cluster stability stats and return the summary ari in a data frame
+  summary_stability_stats_df <-
+    get_cluster_stability_summary(sce,
+                                  nn_range,
+                                  cluster_type) %>%
+    dplyr::mutate(param_value = as.numeric(param_value))
+  
+  # If a second cluster type is provided, check the cluster stability stats for
+  # each of the clusters in the SCE object associated with the second cluster type
+  if (!is.null(second_cluster_type)) {
+    second_summary_stability_stats_df <- 
+      get_cluster_stability_summary(sce,
+                                    nn_range,
+                                    second_cluster_type) %>%
+      dplyr::mutate(param_value = as.numeric(param_value))
+    
+    summary_stability_stats_df <- summary_stability_stats_df %>%
+      dplyr::bind_rows(second_summary_stability_stats_df)
+  }
+}
```

Comment:
```suggestion
  # Check the cluster stability stats and return the summary ari in a data frame
  summary_stability_stats_df_list <- purrr::map(cluster_types, 
    ~ get_cluster_stability_summary(sce,
                                  nn_range,
                                  .x )
 summary_stability_stats_df <- dplyr::bind_rows(summary_stability_stats_df) &>&
    dplyr::mutate(param_value = as.numeric(param_value))
 
```
Rather than have a whole other function you can condense this and use `purrr` here to make a list of dataframes, one for each cluster type and then combine together into one dataframe.